src/test/java/com/fasterxml/jackson/databind/format/NumberFormatTest.java

```diff
@@ -0,0 +1,53 @@
+package com.fasterxml.jackson.databind.format;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Assert;
+
+import java.math.BigDecimal;
+
+import java.util.Locale;
+import java.util.TimeZone;
+
+public class NumberFormatTest extends BaseMapTest
+{
+    protected static class NumberWrapper {
+
+        public BigDecimal value;
+
+        public NumberWrapper() {}
+        public NumberWrapper(BigDecimal v) { value = v; }
+    }
+
+    public void testTypeDefaults() throws Exception
+    {
+        ObjectMapper mapper = newJsonMapper();
+        mapper.configOverride(BigDecimal.class)
+            .setFormat(new JsonFormat.Value("00,000.00", JsonFormat.Shape.STRING, (Locale) null, (TimeZone) null, null, null));
+        String json = mapper.writeValueAsString(new NumberWrapper(new BigDecimal("1234")));
+        assertEquals(a2q("{'value':'01,234.00'}"), json);
+
+        // and then read back is not supported yet.
+        /*NumberWrapper w = mapper.readValue(a2q("{'value':'01,234.00'}"), NumberWrapper.class);
+        assertNotNull(w);
+        assertEquals(new BigDecimal("1234"), w.value);*/
+    }
+
+    protected static class InvalidPatternWrapper {
+        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "#,##0.#.#")
+        public BigDecimal value;
+
+        public InvalidPatternWrapper(BigDecimal value) {
+            this.value = value;
+        }
+    }
+
+    public void testInvalidPattern() throws Exception {
+        ObjectMapper mapper = newJsonMapper();
+        Assert.assertThrows(JsonMappingException.class, () -> {
+            mapper.writeValueAsString(new InvalidPatternWrapper(BigDecimal.ZERO));
+        });
+    }
```

Comment:
```suggestion
    protected static class InvalidPatternWrapper {
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "#,##0.#.#")
        public BigDecimal value;

        public InvalidPatternWrapper(BigDecimal value) {
            this.value = value;
        }
    }

    @Test
    public void testTypeDefaults() throws Exception
    {
        ObjectMapper mapper = newJsonMapper();
        mapper.configOverride(BigDecimal.class)
            .setFormat(new JsonFormat.Value("00,000.00", JsonFormat.Shape.STRING, (Locale) null, (TimeZone) null, null, null));
        String json = mapper.writeValueAsString(new NumberWrapper(new BigDecimal("1234")));
        assertEquals(a2q("{'value':'01,234.00'}"), json);

        // and then read back is not supported yet.
        /*NumberWrapper w = mapper.readValue(a2q("{'value':'01,234.00'}"), NumberWrapper.class);
        assertNotNull(w);
        assertEquals(new BigDecimal("1234"), w.value);*/
    }

    @Test
    public void testInvalidPattern() throws Exception {
        ObjectMapper mapper = newJsonMapper();
        
        try {
            mapper.writeValueAsString(new InvalidPatternWrapper(BigDecimal.ZERO));
            fail();
        } catch (JsonMappingException e) {
            verifyException(e, "expected message part 1/2"); // 
            verifyException(e, "expected message part 2/2");
        } 
    }
```

Couple of suggestion on test if that's okay with you 🙂
1. Let's use JUnit 5, (meaning remove `extends BaseMapTest`
2. Class declaration comes first then test methods (Check suggestion)
3. As suggested above, check message by using `try-catch` and `verifyException` (from `DatabindTestUtil`) combo

Reply:
Above comment seems like [Jackson Coding Style convention](https://github.com/FasterXML/jackson/blob/master/contribution/jackson-coding-style-guide.md#testing-conventions) idea 🤔...  WDYT, @cowtowncoder ?